internal/command/logs/ship.go

```diff
@@ -0,0 +1,113 @@
+package logs
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"github.com/superfly/flyctl/api"
+	"github.com/superfly/flyctl/client"
+	"github.com/superfly/flyctl/flaps"
+	"github.com/superfly/flyctl/flyctl"
+	"github.com/superfly/flyctl/internal/command"
+	"github.com/superfly/flyctl/internal/command/orgs"
+	"github.com/superfly/flyctl/internal/command/secrets"
+	"github.com/superfly/flyctl/internal/flag"
+	"github.com/superfly/flyctl/iostreams"
+)
+
+func newShip() *cobra.Command {
+
+	const (
+		short = "Ship logs to an external provider"
+		long  = short + "\n"
+		usage = "ship"
+	)
+
+	cmd := command.New(usage, short, long, runShip, command.RequireSession)
+
+	flag.Add(cmd)
+
+	cmd.Args = cobra.MaximumNArgs(1)
+
+	return cmd
+}
+
+func runShip(ctx context.Context) (err error) {
+	client := client.FromContext(ctx).API()
+	io := iostreams.FromContext(ctx)
+	selectedOrg, err := orgs.OrgFromFirstArgOrSelect(ctx)
+
+	if err != nil {
+		return err
+	}
+
+	appName := selectedOrg.Slug + "-auto-log-shipper"
+
+	var app *api.AppCompact
+
+	app, err = client.GetAppCompact(ctx, appName)
+
+	if err != nil {
+		input := api.CreateAppInput{
+			Name:           appName,
+			OrganizationID: selectedOrg.ID,
+			Machines:       true,
+		}
+
+		createdApp, err := client.CreateApp(ctx, input)
+		app = client.AppToCompact(createdApp)
+
+		if err != nil {
+			return err
+		}
+	}
+
+	fmt.Fprintf(io.Out, "Setting up secrets for %s\n", app.Name)
```

Comment:
how about using stderr here? that would allow stdout to support the `--json` flag in the future for returning the app information.